src/lib/__tests__/chartStorage.js

```diff
@@ -0,0 +1,310 @@
+const { JSDOM } = require('jsdom'); 
+
+// Mock the Local Storage. 
+
+const dom = new JSDOM('<!doctype html><html><body></body></html>', {
+    url: 'http://localhost'
+});
+global.window = dom.window;
+global.document = dom.window.document;
+global.localStorage = {
+    getItem: jest.fn(),
+    setItem: jest.fn(),
+}; 
+const { saveChart , loadAllSavedCharts , loadSavedChart , updateCurrentChartData , loadCurrentChartData } = require( "../chartStorage" ) ; 
+
```

Comment:
For making this easier, you can actually use:

beforeEach(() => {
    window.localStorage.clear();
});

To not have to include it at the end of every test, and be guaranteed that the storage is cleared.